packages/service-library/src/servicelib/redis/_project_lock.py

```diff
@@ -59,17 +62,20 @@ async def _wrapper(*args: P.args, **kwargs: P.kwargs) -> R:
             )
             async def _exclusive_func(*args, **kwargs) -> R:
                 if notification_cb is not None:
-                    await notification_cb()
+                    with log_catch(_logger, reraise=False):
+                        await notification_cb()
                 return await func(*args, **kwargs)
 
             try:
-                result = await _exclusive_func(*args, **kwargs)
-                # we are now unlocked
-                if notification_cb is not None:
-                    await notification_cb()
-                return result
+                return await _exclusive_func(*args, **kwargs)
+
             except CouldNotAcquireLockError as e:
                 raise ProjectLockError from e
+            finally:
+                # we are now unlocked
+                if notification_cb is not None:
+                    with log_catch(_logger, reraise=False):
```

Comment:
THOUGH: I like the functionality of `log_catch` but not the reraise argument. 
I wonder if we could have something more flexible and readable as
```python
with suppress(SomeException, OtherException), log_raised_exceptions(_logger, logging.WARNING): 
           funtion_that_might_raise_some_exception()
```

Reply:
not sure I find that readable ;)

Reply:
maybe we should have a context manager for this use-case (with timeout and all), instead of concatenating 4 of them. it's not very readable ;)